Makefile doesn't validate $GOPATH exists

# Report

## What did you do?

Run tests.

I tried `make validate` 

⚠️ It actually fails out in the `getdeps` target.

## What did you expect to happen?

I expected it to run validation stuff. 

## What happened instead?
```
/usr/bin/which: no golangci-lint in (/usr/kerberos/bin:/usr/local/bin:/usr/local/sbin:/home/stahnma/bin:/bin:/usr/bin:/sbin:/usr/sbin:/usr/X11R6/bin:/home/stahnma/.gem/ruby/1.8/bin:/home/stahnma/go/bin:/usr/local/go/bin)
Installing golangci-lint
golangci/golangci-lint info checking GitHub for tag 'v1.42.0'
golangci/golangci-lint info found version: 1.42.0 for v1.42.0/linux/amd64
install: cannot create regular file '/bin/golangci-lint': Permission denied
make: *** [Makefile:28: getdeps] Error 1
```



## Your Environment

Please complete the following information):
 - Flottbot version: cc178d3807fbe08f87a488c014df05a270e1b4e7
 - OS: Fedora 34
 - OS version: [e.g. 10.13.6]

## Additional context

First step: The Makefile assumes that `$GOPATH` is set to something. If it's not set, it attempts to write in `/bin` which is, of course, owned by root. This means tests can't run.  


Next I tried hard-setting GOPATH, though I normally don't have to. 
```

[0] stahnma@spike ~/go/src/github.com/target/flottbot [main]> echo $GOPATH
/home/stahnma/go
[0] stahnma@spike ~/go/src/github.com/target/flottbot [main]> make validate
/usr/bin/which: no golangci-lint in (/usr/kerberos/bin:/usr/local/bin:/usr/local/sbin:/home/stahnma/bin:/bin:/usr/bin:/sbin:/usr/sbin:/usr/X11R6/bin:/home/stahnma/.gem/ruby/1.8/bin:/home/stahnma/go/bin:/usr/local/go/bin)
Installing golangci-lint
golangci/golangci-lint info checking GitHub for tag 'v1.42.0'
golangci/golangci-lint info found version: 1.42.0 for v1.42.0/linux/amd64
install: cannot create regular file '/bin/golangci-lint': Permission denied
make: *** [Makefile:28: getdeps] Error 1
```

Finally, 

Looking more in depth at the `Makefile` it looks like the installation argument for `/golangci-lint` is just configured wrong. (needs `shell` in there. PR incoming. 



I would expect either to warn the user and exit, or attempt to install somewhere that isn't global.

--- a/Makefile
+++ b/Makefile
@@ -28,7 +28,7 @@ getdeps:
 	@which golangci-lint 1>/dev/null || \
 		(echo "Installing golangci-lint" && \
 		curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | \
-		sh -s -- -b $(go env GOPATH)/bin $(GOLANGCI_LINT_VERSION))
+		sh -s -- -b $(shell go env GOPATH)/bin $(GOLANGCI_LINT_VERSION))
 
 .PHONY: lint
 lint:

